game-core/src/main/java/games/strategy/net/ServerMessenger.java

```diff
@@ -60,13 +62,57 @@
   private final Map<INode, SocketChannel> nodeToChannel = new ConcurrentHashMap<>();
   private final Map<SocketChannel, INode> channelToNode = new ConcurrentHashMap<>();
 
+  /**
+   * The lobby database if this instance is for use by a lobby or {@code null} if this instance is for use by a game
+   * host.
+   */
+  private final @Nullable Database database;
```

Comment:
During the DI exercise, it became apparent that this class contains behavior for two different kinds of servers: a game host and a lobby.  This can be seen through the repeated usage of the private `isLobby()` method to conditionally execute code throughout the implementation of this class.

By replacing the dependency on the `Database` static methods with a `Database` instance, I had to introduce a new field to store the `Database`, which is `null` for a game host and non-`null` for a lobby.  This makes this class a bit messier than it already was.  Ideally, the lobby-specific logic should be split off into a subclass or something similar.  However, that would be outside the scope of this PR, and so I tried to notify the code reader of this class's dual nature via some new Javadocs on the constructors

Reply:
About expected to run into such issues. Eventually such things will be cleanly split out and won't be in scope of each other at all.

Please double check the javadoc commentary, it's a bit odd to have the 'if' conditional at the end of the sentence after the noun "the lobby database". Can also be simpler perhaps by replacing `{@code null} if this instance is for use by a game host.` -> `otherwise {@code null}`